### DISCUSSION TITLE
Some albums get this error

### DISCUSSION THREAD

**Discussion by apr292000** [s0hv/foo_discord_rich#5] [Q&A] [answered]
Discord Rich Presence Integration v1.3.0-beta+f6c78e4: artwork uploader exited with status: 259 and error: 
Discord Rich Presence Integration v1.3.0-beta+f6c78e4Critical error: File is already in use

**Comment by s0hv** [comment]
What script are you using for uploading the file? Also does the problem always happen for the same albums or is it random (is it refusing to work at all or does it work after retrying)?

**Reply by apr292000** [reply]
I used the script provided, and ye just that whole album wouldn't work.

**Reply by s0hv** [reply]
Is the artwork embedded in the audio file or is it stored separately as an image?

**Reply by apr292000** [reply]
just as cover.png

**Reply by apr292000** [reply]
gonna try downloading different album art to test out, i think i may of converted the album art from a webp to png

**Reply by apr292000** [reply]
Still getting same error in console

**Reply by s0hv** [reply]
There might be some other plugins that lock the file then. You could try adding
 ```py
import time
time.sleep(3)
```
to the top of the script which would make it always wait 3 seconds before uploading the image. If that fixes it then it's most likely that some other plugin is locking the image file so it cannot be read by other processes.

**Reply by apr292000** [reply]
didn't fix it

**Reply by s0hv** [reply]
Are you able to delete the file after switching to the next track on the album? Another option could be to try the rust script I made and see if that works in case you are uploading to imgur https://github.com/s0hv/rust-imgur-upload/releases

**Reply by apr292000** [reply]
after messing around i now only get Discord Rich Presence Integration v1.3.0-beta+f6c78e4: artwork uploader exited with status: 259 and error: and not the other part

**Reply by s0hv** [reply]
So there still an error but no message that tells what the error is? How'd you manage that? Also can you try to upload the artwork by right clicking the track and the from "Discord rich presence" click "generate artwork url".

**Reply by apr292000** [reply]
Tried and it didnt work 
Discord Rich Presence Integration v1.3.0-beta+f6c78e4: artwork uploader exited with status: 259 and error: 
Discord Rich Presence Integration v1.3.0-beta+f6c78e4: 0 entries updated

**Reply by s0hv** [reply]
Uhh this is not easy to debug. Have you done any changes to the script since no error message is shown?

**Reply by apr292000** [reply]
nope all i did was refresh my image cache on eole skin

**Reply by s0hv** [reply]
Do you mind changing your script to something like this. It will log the filepath of the cover and any errors to a file called log.log which will be located in the same directory as the script itself. Then see what gets written to log.log when you try to upload cover art.

```py
import requests
import sys
import os
import logging

url = "https://api.imgur.com/3/image"

dir_path = os.path.dirname(os.path.realpath(__file__))
logging.basicConfig(filename=os.path.join(dir_path, 'log.log'),
                    filemode='a',
                    format='%(asctime)s,%(msecs)d %(name)s %(levelname)s %(message)s',
                    datefmt='%H:%M:%S',
                    level=logging.DEBUG)

try:
    sys.stdin.reconfigure(encoding='utf-8')
    filename = sys.stdin.read().strip()
    logging.info(f'Filename: {filename}')
    client_id = 'Insert imgur api client id here'

    headers = {
        'Authorization': f'Client-ID {client_id}'
    }

    with open(filename, 'rb') as f:
        r = requests.post(url, headers=headers, files={'image': f})

    if not r.ok:
        print(r.text[:1000])
        exit(1)


    print(r.json()['data']['link'], end='')
except:
    logging.exception('Failed to upload')
```

**Reply by apr292000** [reply]
00:57:07,599 root INFO Filename: D:\Music\Kanye West\2010 - My Beautiful Dark Twisted Fantasy\cover.jpg
00:57:07,607 urllib3.connectionpool DEBUG Starting new HTTPS connection (1): api.imgur.com:443
00:57:21,416 urllib3.connectionpool DEBUG https://api.imgur.com:443 "POST /3/image HTTP/1.1" 403 None
00:57:21,424 root ERROR Failed to upload
Traceback (most recent call last):
  File "C:\Users\____\AppData\Roaming\Python\Python39\Scripts\uploader.py", line 30, in <module>
    exit(1)
  File "C:\Users\____\AppData\Local\Programs\Python\Python39\lib\_sitebuiltins.py", line 26, in __call__
    raise SystemExit(code)
SystemExit: 1

**Reply by s0hv** [reply]
Looks like the authorization is invalid and imgur returns 403 Forbidden. This most likely means that there's something wrong with the imgur api client id. Make sure the `client_id` variable has a valid value

**Reply by apr292000** [reply]
![image](https://user-images.githubusercontent.com/24740401/214594865-34f69e1b-c4c4-4535-87cd-2c5e1c82f90a.png)
It's valid it works for other albums

**Reply by s0hv** [reply]
That's weird. Does it work if you try to manually upload the file to imgur from https://imgur.com/upload.

**Comment by apr292000** [comment, marked as answer]
I think i fixed it, for some reason it just hated the file size, downloaded a smaller size album cover and it seems fixed/ or the file itself

**Reply by s0hv** [reply]
Max size for images seems to be 20MB https://help.imgur.com/hc/en-us/articles/115000083326-What-files-can-I-upload-Is-there-a-size-limit- Good to know for future

**Comment by s0hv** [comment]
The reason for error 259 was found. I wrote on ways to fix it in this answer https://github.com/s0hv/foo_discord_rich/discussions/9#discussioncomment-5766113
